contracts/contracts/strategies/LidoWithdrawalStrategy.sol

```diff
@@ -0,0 +1,307 @@
+        // the vault to transfer WETH to it. This returns any deposited WETH
+        // to the vault so that it is not lost for balance tracking purposes.
+        uint256 wethBalance = weth.balanceOf(address(this));
+        if (wethBalance > 0) {
+            // slither-disable-next-line unchecked-transfer
+            weth.transfer(vaultAddress, wethBalance);
+        }
+
+        emit Deposit(address(stETH), address(withdrawalQueue), stETHStart);
+    }
+
+    /**
+     * @notice Withdraw an asset from the underlying platform
+     * @param _recipient Address to receive withdrawn assets
+     * @param _asset Address of the asset to withdraw
+     * @param _amount Amount of assets to withdraw
+     */
+    function withdraw(
+        // solhint-disable-next-line no-unused-vars
+        address _recipient,
+        // solhint-disable-next-line no-unused-vars
+        address _asset,
+        // solhint-disable-next-line no-unused-vars
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        // Does nothing - all withdrawals need to be called manually using the
+        // Strategist calling claimWithdrawals
+        revert("use claimWithdrawals()");
+    }
+
+    /**
+     * @notice Claim previously requested withdrawals that have now finalized.
+     * Called by the Strategist.
+     * @param _requestIds Array of withdrawal request identifiers
+     * @param expectedAmount Total amount of ETH expect to be withdrawn
+     */
+    function claimWithdrawals(
+        uint256[] memory _requestIds,
+        uint256 expectedAmount
+    ) external nonReentrant {
+        require(
+            msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        uint256 startingBalance = payable(address(this)).balance;
+        uint256 lastIndex = withdrawalQueue.getLastCheckpointIndex();
+        uint256[] memory hintIds = withdrawalQueue.findCheckpointHints(
+            _requestIds,
+            1,
+            lastIndex
+        );
+        withdrawalQueue.claimWithdrawals(_requestIds, hintIds);
+
+        uint256 currentBalance = payable(address(this)).balance;
+        uint256 withdrawalAmount = currentBalance - startingBalance;
+        // Withdrawal amount should be within 2 wei of expected amount
+        require(
+            withdrawalAmount + 2 >= expectedAmount &&
+                withdrawalAmount <= expectedAmount,
+            "Withdrawal amount not expected"
+        );
+
+        emit WithdrawalClaims(_requestIds, withdrawalAmount);
+
+        outstandingWithdrawals -= withdrawalAmount;
+        weth.deposit{ value: currentBalance }();
+        // slither-disable-next-line unchecked-transfer
+        weth.transfer(vaultAddress, currentBalance);
+        emit Withdrawal(
+            address(weth),
+            address(withdrawalQueue),
+            currentBalance
+        );
+    }
+
+    function _abstractSetPToken(address, address) internal override {
```

Comment:
Maybe throw this internal method to the bottom of the contract? So it's not between all the external withdraw methods.

Reply:
done. I've also made it pure